src/cartesi-machine.lua

```diff
@@ -1923,47 +1924,12 @@ local htif_yield_mode = {
     [cartesi.machine.HTIF_YIELD_CMD_AUTOMATIC] = "Automatic",
 }
 
-local function is_power_of_two(value) return value > 0 and ((value & (value - 1)) == 0) end
-
-local function ilog2(value)
-    value = assert(math.tointeger(value), "expected integer")
-    assert(value ~= 0, "expected non-zero integer")
-    local log = 0
-    while value > 1 do
-        log = log + 1
-        value = value >> 1
-    end
-    return log
-end
-
-local function check_cmio_memory_range_config(range, name)
-    assert(range, string.format("cmio range %s must be defined", name))
-    assert(not range.shared, string.format("cmio range %s cannot be shared", name))
-    assert(
-        is_power_of_two(range.length),
-        string.format("cmio range %s length not a power of two (%u)", name, range.length)
-    )
-    local log = ilog2(range.length)
-    local aligned_start = (range.start >> log) << log
-    assert(
-        aligned_start == range.start,
-        string.format("cmio range %s start not aligned to its power of two size", name)
-    )
-    range.image_filename = nil
-end
-
 local function check_cmio_htif_config(htif)
     assert(not htif.console_getchar, "console getchar must be disabled for cmio")
     assert(htif.yield_manual, "yield manual must be enabled for cmio")
     assert(htif.yield_automatic, "yield automatic must be enabled for cmio")
 end
 
-local function set_yield_data(machine, reason, data)
-    local m16 = (1 << 16) - 1
-    local m32 = (1 << 32) - 1
-    machine:write_htif_fromhost_data(((reason & m16) << 32) | (data & m32))
-end
-
 local function get_yield(machine)
```

Comment:
Since we could get rid of `set_yield_data`, I think it would also make sense to get rid of this method in the future (another issue in this case).

Just because I think the amount of bit shifting here is fragile.